pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -0,0 +1,443 @@
+				return err
+			}
+			c.installedActiveGWName = activeGW.Name
+			return c.addMCFlowsForCIImps(activeGW)
+		}
+	}
+	return nil
+}
+
+func (c *MCRouteController) syncInterClusterFlows(activeGW *mcv1alpha1.Gateway) error {
+	installedCIImports := c.installedCIImports.List()
+	desiredCIImports, err := c.ciImportLister.ClusterInfoImports(c.namespace).List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	// We do a full sync here anyway for all desired ClusterInfoImports.
+	if err := c.addMCFlowsForCIImps(activeGW); err != nil {
+		return err
+	}
+
+	for _, ciImp := range installedCIImports {
+		ciImport := ciImp.(*mcv1alpha1.ClusterInfoImport)
+		if !c.containsCIImport(desiredCIImports, ciImport) {
+			if err := c.deleteMCNodeFlowsForCIImport(ciImport, activeGW.Name); err != nil {
+				return err
+			}
+			c.installedCIImports.Delete(ciImport)
+		}
+	}
+	return nil
+}
+
+func (c *MCRouteController) containsCIImport(ciImports []*mcv1alpha1.ClusterInfoImport, ciImport *mcv1alpha1.ClusterInfoImport) bool {
+	for _, cii := range ciImports {
+		if cii.Name == ciImport.Name {
+			return true
+		}
+	}
+	return false
+}
+
+func (c *MCRouteController) cleanupFlows(gatewayName string) error {
+	if err := c.deleteMCNodeFlowsForGateway(gatewayName); err != nil {
+		return err
+	}
+	if err := c.ofClient.UninstallMulticlusterFlows(gatewayName); err != nil {
+		return err
+	}
+	klog.V(2).InfoS("Deleted flows for Gateway", "gateway", gatewayName)
+	return nil
+}
+
+func (c *MCRouteController) addMCFlowsForCIImps(activeGW *mcv1alpha1.Gateway) error {
+	allciImports, err := c.ciImportLister.ClusterInfoImports(c.namespace).List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	if len(allciImports) == 0 {
+		klog.V(2).InfoS("No remote ClusterInfo imported, do nothing")
+		return nil
+	}
+
+	for _, ciImport := range allciImports {
+		if err := c.addMCFlowsForSingleCIImp(activeGW, ciImport); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *MCRouteController) addMCFlowsForSingleCIImp(activeGW *mcv1alpha1.Gateway, ciImport *mcv1alpha1.ClusterInfoImport) error {
+	tunnelPeerIP := getPeerGatewayIP(ciImport.Spec)
+	klog.InfoS("Adding/updating remote Gateway Node flows for Multi-cluster", "gateway", klog.KObj(activeGW),
+		"node", c.nodeConfig.Name, "peer", tunnelPeerIP)
+	allCIDRs := []string{ciImport.Spec.ServiceCIDR}
+	peerConfigs, err := generatePeerConfigs(allCIDRs, tunnelPeerIP)
+	if err != nil {
+		klog.ErrorS(err, "Parse error for serviceCIDR from remote cluster", "clusterinfoimport", ciImport.Name, "gateway", activeGW.Name)
+		return err
+	}
+	if activeGW.Name == c.nodeConfig.Name {
+		klog.V(2).InfoS("Adding/updating flows to remote Gateway Node for Multi-cluster traffic", "clusterinfoimport", ciImport.Name, "cidrs", allCIDRs)
+		localGatewayIP := net.ParseIP(activeGW.GatewayIP)
+		if err := c.ofClient.InstallMulticlusterGatewayFlows(
+			activeGW.Name+ciImport.Name,
```

Comment:
I think we don't need `activeGW.Name` in the name? ciImport's uniqueness can guarantee the flow's uniqueness and we don't have more than one active gateway, right?

Reply:
This is exactly my question too.

Reply:
yeah, after double checking the code, I removed them. thanks.